### PR TITLE
New blog post for cellAlias

### PR DESCRIPTION
**Pull request by pratikgupta123**
This blog post outlines the configuration for cellAlias to allow cross cell replica reads by VTgate.

### CODE REVIEW

**Comment by pratikgupta123** [comment]
@morgo Request you to please review this.

**Comment by deepthi** [comment]
Welcome @pratikgupta123. We appreciate the contribution!

File: content/en/blog/2020-05-19-vitess-cross-cell.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+---
+author: "Pratik Gupta"
+date: 2020-05-19T11:20:00
+slug: "2020-05-19-vitess-cross-cell"
+tags: ['vitess', 'vitess cluster', 'vitess cross cell', 'vitess getting started']
+title: "Vitess: The Cross Cell Connection"
+---
+
+This post explains about how VTGate handles cross-cell operations and how to configure CellAlias for cross cell read operations.
```
Comment (deepthi) [inline content/en/blog/2020-05-19-vitess-cross-cell.md]: It is good to be consistent with phrasing. `cross-cell` everywhere or `cross cell` everywhere but not both.

File: content/en/blog/2020-05-19-vitess-cross-cell.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+If you are new to Vitess, it is recommended to [read this blog post](./2020-04-27-life-of-a-cluster) to get more familiar with the various components and its configuration in Vitess.
```
Comment (deepthi) [inline content/en/blog/2020-05-19-vitess-cross-cell.md]:
```suggestion
If you are new to Vitess, it is recommended to [read this blog post](./2020-04-27-life-of-a-cluster) to get more familiar with the various components and their configuration in Vitess.
```

File: content/en/blog/2020-05-19-vitess-cross-cell.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+To understand CellAlias, first let's get familiar with what a cell means in Vitess. A cell is a group of servers and network infrastructure collocated in an area, and isolated from failures in other cells. It is typically either a full data center or a subset of a data center, sometimes called a zone or availability zone. Vitess gracefully handles cell-level failures, such as when a cell is cut off the network.
+By default Vitess limits cross cell traffic by only routing the writes to a master if it does not reside in the current cell. Vitess will always prefer the replicas in the current cell for all read operations.
+
+However in some cases it may be necessary to route read operations to different cells eg. when all the replicas in the current cell fail. For all such cases, we need to configure CellAlias.
+
+## CellAlias
+
+A CellAlias defines a group of cells within which replica/rdonly traffic can be routed across cells. By default, Vitess does not allow traffic between replicas that are part of different cells. Between cells that are not in the same group (alias), only master traffic can be routed.
```
Comment (deepthi) [inline content/en/blog/2020-05-19-vitess-cross-cell.md]:
```suggestion
A CellAlias defines a group of cells within which replica/rdonly traffic can be routed across cells. By default, Vitess does not allow replica traffic between different cells. Between cells that are not in the same group (alias), only master traffic can be routed.
```

File: content/en/blog/2020-05-19-vitess-cross-cell.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+vtctl \
+  -topo_implementation=etcd2 \
+  -topo_global_server_address=localhost:2379 \
+  -topo_global_root=/vitess/global \
+  AddCellsAlias \
+  -cells zone1,zone2 \
+  cellglobal
+```
+
+Here `zone1` an `zone2` are two cells in different datacenters and `cellglobal` is the alias name.
```
Comment (deepthi) [inline content/en/blog/2020-05-19-vitess-cross-cell.md]:
```suggestion
Here `zone1` and `zone2` are two cells in different datacenters and `cellglobal` is the alias name.
```

File: content/en/blog/2020-05-19-vitess-cross-cell.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  -mysql_auth_server_impl=none
+```
+
+### Understanding VTGate Cross-Cell Behaviour
+
+1. VTGate will always be able to route to a master in a different cell (which is part of cells_to_watch) even without a cellAlias. the cellAlias requirement is only for replicas.
+2. By default VTGate sends all queries(reads and writes) to the master.
+3. Replica reads can be performed by first issuing a `use ks@replica` where `ks` should be replaced by your keyspace name.
+4. VTGate will always prefer tables that are in the same cell over tablets in other cells despite the CellAlias.
+5. VTGate caches the CellAliases from the Topology Server, so if an alias is created/updated after the VTGate instance has been started then VTGate must be reloaded to fetch the CellAlias. This behaviour is similar to adding/removing new cells where the VTGate instances need to be restarted to udpate the `-cells_to_watch` values.
```
Comment (deepthi) [inline content/en/blog/2020-05-19-vitess-cross-cell.md]:
```suggestion
5. VTGate caches the CellAliases from the Topology Server, so if an alias is created/updated after the VTGate instance has been started then VTGate must be restarted to fetch the CellAlias. This behaviour is similar to adding/removing new cells where the VTGate instances need to be restarted to udpate the `-cells_to_watch` values.
```

**Review by deepthi — COMMENTED** [review]
Very nicely written and comprehensive.
A few minor suggestions/typos.

**Comment by pratikgupta123** [comment]
> Very nicely written and comprehensive.
> A few minor suggestions/typos.

Thank you @deepthi 
I agree with the suggestions and will commit them as a single change.

**Comment by pratikgupta123** [comment]
@deepthi , @morgo 
I just could not get around the failing DCO check after accepting the review changes.

Let me know if it is ok, else I am tempted to close this PR and create another one with much cleaner commits.

**Comment by deepthi** [comment]
> @deepthi , @morgo
> I just could not get around the failing DCO check after accepting the review changes.
> 
> Let me know if it is ok, else I am tempted to close this PR and create another one with much cleaner commits.

I see that you signed-off all the commits, so I suspect the problem is the suggestions I made in the review. A new PR might be simplest, but please feel free to do whatever is easiest for you.

**Comment by pratikgupta123** [comment]
@deepthi 
I will close this one and create another one so it has much cleaner commits.

**Pull request closed without merging**